Spigot-API-Patches/0067-Added-ban-events.patch

```diff
@@ -0,0 +1,182 @@
+From c733f27c391faff1ffe6b562194afa4f45abb901 Mon Sep 17 00:00:00 2001
+From: Sam Ezeh <spammy23@noreply.users.github.com>
+Date: Wed, 30 Aug 2017 21:57:51 +0100
+Subject: [PATCH] Added ban events
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/server/BanEvent.java b/src/main/java/com/destroystokyo/paper/event/server/BanEvent.java
+new file mode 100644
+index 00000000..bd9f66af
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/server/BanEvent.java
+@@ -0,0 +1,132 @@
++package com.destroystokyo.paper.event.server;
++
++import org.bukkit.BanEntry;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.server.ServerEvent;
++
++import java.util.Date;
++
++/**
++ * Called when a player is banned
++ */
++public abstract class BanEvent extends ServerEvent implements Cancellable {
```

Comment:
I don't see a reason to extend ServerEvent here. Why does that class even exists...

Especially considering cases of this event being triggered may be player oriented, not server.

Which adds next request. Replace getSource and all usages of these strings with the CommandSender instead.  a string is not accurate enough to determine the proper sender (IE: For Command Blocks or Console Sender)

the source string should not be exposed.

I'm also on the fence about target. Id rather Profile oriented bans return an OfflinePlayer, and IP bans give the IP in string or maybe a representative Java Object (InetAddress I think it is?)